### PR TITLE
First version of the feature send icons

### PR DESCRIPTION
**Pull request by el-paso**
With this feature you can send 8x8 icons to the LED matrix.

### CODE REVIEW

**Comment by el-paso** [comment]
With this feature you can send 8x8 icons to the LED matrix.

**Comment by el-paso** [comment]
Please review and merge @TheNitek

File: web/index.html [hunk 1]
```diff
@@ -52,6 +53,41 @@
       form {
         float: left;
       }
+      .tg  {
```
Comment (TheNitek) [inline web/index.html]: tg?

File: web/index.html [hunk 2]
```diff
@@ -52,6 +53,41 @@
...
+        border-color:black;
+        border-style:solid;
+        border-width:1px;
+        font-family:Arial, sans-serif;
+        font-size:14px;
+        overflow:hidden;
+        padding:10px 5px;
+        word-break:normal;
+      }
+      .tg th {
```
Comment (TheNitek) [inline web/index.html]: Are there any THs?

File: web/index.html [hunk 3]
```diff
@@ -52,6 +53,41 @@
...
+        overflow:hidden;
+        padding:10px 5px;
+        word-break:normal;
+      }
+      .tg .tg-73oq { 
+        border-color:#000000;
+        text-align:left;
+        vertical-align:top
+      }
+      img {
```
Comment (TheNitek) [inline web/index.html]: Should probably be a class

File: web/index.html [hunk 4]
```diff
@@ -253,6 +289,72 @@ <h1>Draw something</h1>
...
+        var c = document.getElementById("myCanvas");
+        var ctx = c.getContext("2d");
+        base_image = new Image();
+        base_image.src = iconPath;
+        base_image.onload = function(){
+          ctx.drawImage(base_image, 0, 0);
+          var d = ctx.getImageData(0, 0, 8, 8).data;
+
+          for (let i = 0; i < (d.length / 4); i++) {
+            const element = d[i];
```
Comment (TheNitek) [inline web/index.html]: Never used?

File: web/index.html [hunk 5]
```diff
@@ -253,6 +289,72 @@ <h1>Draw something</h1>
...
+            b = d[(i*4)+2];
+            let p = ((r & 0xF8) << 8) | ((g & 0xFC) << 3) | (b >> 3);
+
+            let payload = new Uint8Array(3);
+            payload[0] = (x << 4) | y;
+            payload[1] = p >> 8;
+            payload[2] = p & 0xFF;
+            message = new Paho.MQTT.Message(payload);
+            message.destinationName = DRAW_TOPIC;
+            client.send(message);
```
Comment (TheNitek) [inline web/index.html]: No need to send this pixel by pixel. The matrix can be updated using one 128byte message

File: web/index.html [hunk 6]
```diff
@@ -253,6 +289,72 @@ <h1>Draw something</h1>
...
+
+        }
+
+      }
+
     </script>
+    <br>
+    <br>
+    <h1>Send some icons</h1>
+    <center>
```
Comment (TheNitek) [inline web/index.html]: Deprecated

File: web/index.html [hunk 7]
```diff
@@ -253,6 +289,72 @@ <h1>Draw something</h1>
...
+            message.destinationName = DRAW_TOPIC;
+            client.send(message);
+          }
+
+        }
+
+      }
+
     </script>
+    <br>
```
Comment (TheNitek) [inline web/index.html]: Why interrupt the <script> with pieces of HTML? Putting the HTML with the rest is probably more readable

File: web/index.html [hunk 8]
```diff
@@ -253,6 +289,72 @@ <h1>Draw something</h1>
...
+  var cell = [];
+
+  for (let i = 0; i < icons.length; i++) {
+    if ((i%3) == 0) {
+      rowCount++;
+      var row = table.insertRow(rowCount);
+      cellCount = 0;
+    }
+    var cell = row.insertCell(cellCount);
+    cell.innerHTML = "<picture><img src=\""+icons[i].iconPath+"\" alt=\""+icons[i].iconTitle+"\" style=\"width:64px;height:64px;\" onclick=\"sendIcon('"+icons[i].iconPath+"')\"></picture>";
```
Comment (TheNitek) [inline web/index.html]: move style into class from line 86?

File: web/index.html [hunk 9]
```diff
@@ -253,6 +289,72 @@ <h1>Draw something</h1>
...
+
     </script>
+    <br>
+    <br>
+    <h1>Send some icons</h1>
+    <center>
+    <table class="tg" id="iconTable"></table>
+    </center>
+
+    <canvas id="myCanvas" width="8" height="8" style="border:1px solid #ff0000; display: none;">
```
Comment (TheNitek) [inline web/index.html]: myCanvas seems a bit generic

why have a border with display: none?

File: web/index.html (hunk 8 above)
Comment (TheNitek) [inline web/index.html]: Probably no need for <picture> tag?

**Comment by TheNitek** [comment]
What's the license of the png files?

**Review by TheNitek — CHANGES_REQUESTED** [review]

File: web/index.html (hunk 9 above)
Comment (el-paso) [reply web/index.html]: Changed it to ```iconPlaceholder```

File: web/index.html (hunk 8 above)
Comment (el-paso) [reply web/index.html]: Done

File: web/index.html (hunk 7 above)
Comment (el-paso) [reply web/index.html]: Changed it

File: web/index.html (hunk 5 above)
Comment (el-paso) [reply web/index.html]: @TheNitek how does the message has to be formatted?

File: web/index.html (hunk 3 above)
Comment (el-paso) [reply web/index.html]: Done

File: web/index.html (hunk 4 above)
Comment (el-paso) [reply web/index.html]: Deleted the line

File: web/index.html (hunk 5 above)
Comment (TheNitek) [reply web/index.html]: https://github.com/TheNitek/draw/blob/f4ccb2c3e06d3de02b21f85f4c60568ee3ac5640/web/index.html#L123

File: web/index.html (hunk 1 above)
Comment (el-paso) [reply web/index.html]: renamed it to iconTable

File: web/index.html (hunk 2 above)
Comment (el-paso) [reply web/index.html]: deleted.

File: web/index.html (hunk 6 above)
Comment (el-paso) [reply web/index.html]: Done

File: web/index.html (hunk 8 above)
Comment (el-paso) [reply web/index.html]: Done

**Comment by el-paso** [comment]
> What's the license of the png files?

I have changed the pictures to versions with  CC0 license.

File: web/index.html (hunk 5 above)
Comment (el-paso) [reply web/index.html]: Done

**Comment by el-paso** [comment]
@TheNitek . All requested changes are implemented. Please review and merge.

**Pull request merged**